roles/foreman_installer/tasks/main.yml

```diff
@@ -0,0 +1,28 @@
+---
+- name: Check that necessary variables are defined
+  assert:
+    that:
+      - foreman_installer_scenario is defined
+
+- name: Check if initial install has happened
+  stat:
+    path: /etc/foreman-installer/scenarios.d/.installed
+  register: installed
+
+- name: 'Run installer in noop mode'
+  command: >
+    {{ foreman_installer_command }}
+    --scenario {{ foreman_installer_scenario }}
+    --noop
+  register: foreman_installer_noop
+  changed_when: false
+  when: installed.stat.exists
+
+- name: 'Run installer' # noqa 301
+  command: >
+    {{ foreman_installer_command }}
+    --scenario {{ foreman_installer_scenario }}
+    {{ (foreman_installer_verbose|bool) | ternary("-v", "") }}
+    {{ (foreman_installer_no_colors|bool) | ternary("--no-colors", "") }}
+    {{ foreman_installer_options | join(' ') }}
+  when: foreman_installer_noop.rc is not defined or foreman_installer_noop.rc != 0
```

Comment:
Should we check for `rc == 2` as in "changes have been made"? Or is that something that gets lost in noop mode?

Reply:
From the noop mode run? I think `noop` returns a `2` if changes will be made.

Reply:
Ohh, maybe you are saying just to be more explicit here. Only run this if its a 2

Reply:
Yes, that's what I was saying

Reply:
Granted, given we don't allow anything but 0 or 2 in the previous step, it won't happen, but I'd still prefer to be explicit.

Reply:
However, given the current untested state of noop, and all of (https://github.com/theforeman/kafo/blob/master/README.md#exit-code) I am inclined to stick with, if `noop` does not return 0 let's run the installer. This should help prevent any cases right now where things in complex scenarios might leave users confused.

Reply:
We've written `  failed_when: foreman_installer_noop.rc not in [0, 2]` above, so the execution will be aborted if the installer exits anything but 0 or 2 and we never reach this step.

Reply:
Hrmm, you are right. You did have me do that. Fair enough.

Reply:
Updated to what I think we landed on design wise